Some tools not executable if image not run as root

```
$ docker run -it --rm -v `pwd`:/project -v /tmp/tmp-phpqa:/tmp -w /project jakzal/phpqa:alpine phpstan analyze
Note: Using configuration file /project/phpstan.neon.dist.
 10/10 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

(...)
                                                                                                                        
 [ERROR] Found 5 errors                                                                                                 
                                                                                                                        

$ docker run -it --rm -v `pwd`:/project -v /tmp/tmp-phpqa:/tmp -w /project -u `id -u`:`id -g` jakzal/phpqa:alpine phpstan analyze
/usr/local/bin/docker-php-entrypoint: exec: line 9: phpstan: Permission denied
```

It's because tools are embedded in `/root` which is not world-readable.

--- a/Dockerfile
+++ b/Dockerfile
@@ -3,11 +3,12 @@ FROM php:7.2-cli
 LABEL maintainer="Jakub Zalas <jakub@zalas.pl>"
 
 ENV COMPOSER_ALLOW_SUPERUSER 1
+ENV COMPOSER_HOME=/tools/.composer
 ENV BUILD_DEPS="autoconf file g++ gcc libc-dev make pkg-config re2c unzip"
 ENV LIB_DEPS="zlib1g-dev"
 ENV TOOL_DEPS="git graphviz"
-ENV PATH="$PATH:/root/.composer/vendor/bin:/root/QualityAnalyzer/bin:/root/DesignPatternDetector/bin:/root/EasyCodingStandard/bin"
-ENV TOOLS_JSON=/root/tools.json
+ENV PATH="$PATH:/tools/.composer/vendor/bin:/tools/QualityAnalyzer/bin:/tools/DesignPatternDetector/bin:/tools/EasyCodingStandard/bin"
+ENV TOOLS_JSON=/tools/tools.json
 
 COPY tools.json ${TOOLS_JSON}
 COPY tools.php /usr/local/bin/tools.php

--- a/Dockerfile-alpine
+++ b/Dockerfile-alpine
@@ -3,11 +3,12 @@ FROM php:7.2-alpine
 LABEL maintainer="Jakub Zalas <jakub@zalas.pl>"
 
 ENV COMPOSER_ALLOW_SUPERUSER 1
+ENV COMPOSER_HOME=/tools/.composer
 ENV BUILD_DEPS="autoconf file g++ gcc libc-dev make pkgconf re2c unzip"
 ENV LIB_DEPS="zlib-dev"
 ENV TOOL_DEPS="git graphviz"
-ENV PATH="$PATH:/root/.composer/vendor/bin:/root/QualityAnalyzer/bin:/root/DesignPatternDetector/bin:/root/EasyCodingStandard/bin"
-ENV TOOLS_JSON=/root/tools.json
+ENV PATH="$PATH:/tools/.composer/vendor/bin:/tools/QualityAnalyzer/bin:/tools/DesignPatternDetector/bin:/tools/EasyCodingStandard/bin"
+ENV TOOLS_JSON=/tools/tools.json
 
 COPY tools.json ${TOOLS_JSON}
 COPY tools.php /usr/local/bin/tools.php

--- a/README.md
+++ b/README.md
@@ -140,11 +140,11 @@ docker run --init -it --rm -v $(pwd):/project -w /project foo/phpqa phpmetrics .
 
 ### Adding PHPStan extensions
 
-A number of PHPStan extensions is available on the image in `/root/.composer/vendor-bin/phpstan/vendor` out of the box.
+A number of PHPStan extensions is available on the image in `/tools/.composer/vendor-bin/phpstan/vendor` out of the box.
 You can find them with the command below:
 
 ```
-phpqa find /root/.composer/vendor-bin/phpstan/vendor/ -iname 'rules.neon' -or -iname 'extension.neon'
+phpqa find /tools/.composer/vendor-bin/phpstan/vendor/ -iname 'rules.neon' -or -iname 'extension.neon'
 ```
 
 Use the composer-bin-plugin to install any additional PHPStan extensions in the `phpstan` namespace:
@@ -155,11 +155,11 @@ FROM jakzal/phpqa:alpine
 RUN global bin phpstan require phpstan/phpstan-phpunit
 ```
 
-You'll be able to include them in your PHPStan configuration from the `/root/.composer/vendor-bin/phpstan/vendor` path:
+You'll be able to include them in your PHPStan configuration from the `/tools/.composer/vendor-bin/phpstan/vendor` path:
 
 ```yaml
 includes:
-    - /root/.composer/vendor-bin/phpstan/vendor/phpstan/phpstan-phpunit/extension.neon
+    - /tools/.composer/vendor-bin/phpstan/vendor/phpstan/phpstan-phpunit/extension.neon
 ```
 
 ## Debugger & Code Coverage

--- a/tools.php
+++ b/tools.php
@@ -171,7 +171,7 @@ namespace Model {
         public function __toString(): string
         {
             return sprintf(
-                'cd $HOME && git clone %s && cd $HOME/%s && git checkout %s && composer install --no-dev --no-suggest --prefer-dist -n && box build && mv %s %s && chmod +x %s && cd && rm -rf $HOME/%s',
+                'cd /tools && git clone %s && cd /tools/%s && git checkout %s && composer install --no-dev --no-suggest --prefer-dist -n && box build && mv %s %s && chmod +x %s && cd && rm -rf /tools/%s',
                 $this->repository,
                 $this->targetDir(),
                 $this->version ?? '$(git describe --tags $(git rev-list --tags --max-count=1) 2>/dev/null)',
@@ -209,7 +209,7 @@ namespace Model {
         public function __toString(): string
         {
             return sprintf(
-                'cd $HOME && git clone %s && cd $HOME/%s && git checkout %s && composer install --no-dev --no-suggest --prefer-dist -n',
+                'cd /tools && git clone %s && cd /tools/%s && git checkout %s && composer install --no-dev --no-suggest --prefer-dist -n',
                 $this->repository,
                 $this->targetDir(),
                 $this->version ?? '$(git describe --tags $(git rev-list --tags --max-count=1) 2>/dev/null)'

